File: 프로그래머스/2/42586. 기능개발/기능개발.java
import java.util.ArrayList;
import java.util.List;

class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        int n = progresses.length;
        int[] days = new int[n];

        for (int i = 0; i < n; i++) {
            int remaining = 100 - progresses[i];
            days[i] = (remaining + speeds[i] - 1) / speeds[i];
        }

        List<Integer> answer = new ArrayList<>();

        int releaseDay = days[0];
        int count = 1;

        for (int i = 1; i < n; i++) {
            if (days[i] <= releaseDay) {
                count++;
            } else {
                answer.add(count);
                releaseDay = days[i];
                count = 1;
            }
        }

        answer.add(count);

        return answer.stream()
                .mapToInt(Integer::intValue)
                .toArray();
    }
}

File: 프로그래머스/2/42586. 기능개발/기능개발.py
def solution(progresses, speeds):
    answer = []
    days = 0
    current = 0
    
    while True:
        days += 1
        if current == len(progresses):
            break
            
        count = 0
        while current < len(progresses) and progresses[current] + speeds[current] * days >= 100:
            current += 1
            count += 1
    
        if count > 0:
            answer.append(count)
    
    return answer
